ruby/src/IceRuby/Communicator.cpp

```diff
@@ -253,18 +219,13 @@ IceRuby_initialize(int argc, VALUE* argv, VALUE /*self*/)
             //
             // We start at index 1 in order to skip the element that we inserted earlier.
             //
-            for (i = 1; i < ac; ++i)
+            for (size_t i = 1; i < seq.size(); ++i)
```

Comment:
We should not do all this args manipulation in the C++ initialize, but move it into the Ruby code. For a follow-up PR.